src/main/java/frc/robot/commands/ScoreSpeakerMovingCommand.java

```diff
@@ -0,0 +1,217 @@
+      TRANSLATION_RATE_LIMIT.in(MetersPerSecondPerSecond), ROTATION_RATE_LIMIT.in(RadiansPerSecond.per(Second)));
+
+  // Reusable objects to prevent reallocation (to reduce memory pressure)
+  private final ChassisSpeeds chassisSpeeds = new ChassisSpeeds();
+  private final MutableMeasure<Angle> turretYawTarget = MutableMeasure.zero(Rotations);
+
+  private Translation2d speakerTranslation;
+
+  private final SwerveRequest.FieldCentricFacingAngle swerveRequestFacing = new SwerveRequest.FieldCentricFacingAngle()
+    .withDriveRequestType(DriveRequestType.Velocity)
+    .withSteerRequestType(SteerRequestType.MotionMagic)
+    .withVelocityX(0.0)
+    .withVelocityY(0.0);
+  
+  private final SwerveRequest.FieldCentric swerveRequestRotation = new SwerveRequest.FieldCentric()
+    .withDriveRequestType(DriveRequestType.Velocity)
+    .withSteerRequestType(SteerRequestType.MotionMagic);
+
+  private boolean isShooting = false;
+
+  public ScoreSpeakerMovingCommand(CommandSwerveDrivetrain drivetrain, ShooterSubsystem shooter,
+      TurretSubsystem turretSubsystem, LEDSubsystem ledSubsystem, BooleanSupplier turretIsSafe) {
+    this.drivetrain = drivetrain;
+    this.shooter = shooter;
+    this.turretSubsystem = turretSubsystem;
+    this.ledSubsystem = ledSubsystem;
+    this.turretIsSafe = turretIsSafe;
+
+    addRequirements(shooter, turretSubsystem);
+
+    swerveRequestFacing.ForwardReference = ForwardReference.RedAlliance;
+    swerveRequestFacing.HeadingController = new PhoenixPIDController(THETA_kP, THETA_kI, THETA_kD);
+    swerveRequestFacing.HeadingController.enableContinuousInput(-PI, PI);
+    swerveRequestFacing.HeadingController.setTolerance(AIM_TOLERANCE.in(Radians));
+
+    swerveRequestFacing.ForwardReference = ForwardReference.RedAlliance;
+  }
+
+  @Override
+  public void initialize() {
+    swerveRequestFacing.HeadingController.reset();
+    shootTimer.reset();
+    rateLimiter.reset(drivetrain.getCurrentFieldChassisSpeeds());
+    var alliance = DriverStation.getAlliance();
+    speakerTranslation = (alliance.isEmpty() || alliance.get() == Blue) ? SPEAKER_BLUE : SPEAKER_RED;
+    isShooting = false;
+  }
+
+  @Override
+  public void execute() {
+    var robotPose = drivetrain.getState().Pose;
+    var robotTranslation = robotPose.getTranslation();
+
+    // Distance between the robot and the speaker
+    var distanceToSpeaker = robotTranslation.getDistance(speakerTranslation);
+
+    // Lookup shooter settings for this distance
+    var shootingSettings = SHOOTER_INTERPOLATOR.calculate(distanceToSpeaker);
+
+    // Calculate time to hit speaker
+    var timeUntilScored = SHOOTER_COEFFICIENT * distanceToSpeaker * SHOOTER_INTERPOLATOR.calculate(distanceToSpeaker).getVelocity().in(RotationsPerSecond);
+
+    // Calculate the predicted offset of the speaker compared to current pose (in meters)
+    var speakerPredictedOffset = new Translation2d((drivetrain.getState().speeds.vxMetersPerSecond * timeUntilScored), 
```

Comment:
I think you want to use `driveTrain.getCurrentFieldChassisSpeeds()` because you want the field relative chassis speeds.